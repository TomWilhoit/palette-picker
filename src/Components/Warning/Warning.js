import React from "react";
import PropTypes from "prop-types";

export const Warning = props => <h2 className="warning">{props.message}</h2>;

Warning.propTypes = {
  message: PropTypes.string,
};