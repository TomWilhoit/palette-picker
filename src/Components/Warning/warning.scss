.warning {
  @include top-right-message;
}