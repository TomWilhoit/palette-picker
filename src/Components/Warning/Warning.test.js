import React from "react";
import { Warning } from "./Warning";
import { shallow } from "enzyme";

describe("Warning", () => {
  let wrapper;
  let mockMessage;
  let props;
  
  beforeEach(() => {
    mockMessage = "Error or Loading message"
    props = {
      message: mockMessage
    };
    wrapper = shallow(<Warning {...props} />);
  })

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  })

  it("should match the snapshot with no message", () => {
    wrapper.setProps({ message: "" });
    expect(wrapper).toMatchSnapshot();
  })

  it("should return an element", () => {
    const expected = (
      <h2 className="warning" />
    );
    let result = Warning(mockMessage);
    expect(result).toEqual(expected);
  })
})