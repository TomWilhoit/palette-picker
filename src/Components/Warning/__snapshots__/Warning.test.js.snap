// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning should match the snapshot with all data passed in 1`] = `
<h2
  className="warning"
>
  Error or Loading message
</h2>
`;

exports[`Warning should match the snapshot with no message 1`] = `
<h2
  className="warning"
/>
`;
